### PR TITLE
 [Ip Camera] Refactor labels and defaults for  openHAB 3

### PR DESCRIPTION
**Pull request by Skinah**
CheckStyle fixes.
Channel label improvements for better 'create equipment from thing' experience in V3.
Readme updated for the above change.
Remove openhab2 from a few places, one caused issues due to folder location not existing in V3.

Signed-off-by: Matthew Skinner matt@pcmus.com

### CODE REVIEW

**Comment by Skinah** [comment]
@fwolter 
You requested/suggested I take a look at `target/code-analysis/report.html` so this PR has done that and fixed up some issues I have noticed during testing under V3

Also what happens if I update the readme.md to better suit V3 users? Does the docs on the website allow the previous version to be viewed using the drop down list that is located in the top right hand corner?

If I select 2.5.x it does not show the current readme as I was hoping, probably as this binding was merged into 2.5.9 and did not exist in 2.5.0
See:
https://www.openhab.org/v2.5/addons/bindings/ipcamera/

**Comment by fwolter** [comment]
> If I select 2.5.x it does not show the current readme as I was hoping, probably as this binding was merged into 2.5.9 and did not exist in 2.5.0

I can't answer this question, but it seems to be a general issue. As the purpose of the drop down field is to have dedicated readmes for different versions, I guess you can go for it and add some OH3 specific parts.

File: bundles/org.openhab.binding.ipcamera/README.md [hunk 1]
```diff
@@ -662,197 +675,48 @@ end
...
-Switch GenericMotionAlarm "Motion detected" { channel="ipcamera:generic:TestCam:motionAlarm" }
-Dimmer GenericAudioThreshold "Audio Threshold [%d]" { channel="ipcamera:generic:TestCam:thresholdAudioAlarm" }
-Switch GenericAudioAlarm "Audio detected" { channel="ipcamera:generic:TestCam:audioAlarm" }
-
-String OutsideCameraGroupHlsUrl "Outside Cameras" { channel="ipcamera:group:OutsideCameras:hlsUrl", ga="Camera" [ protocols="hls" ] }
-Switch OutsideCameraGroupStartHLS "Start outside HLS" { channel="ipcamera:group:OutsideCameras:startStream" }
-
-```
+Use the following example to base your sitemap on to save some time.
+In openHAB 3, if you use the `Create Equipment from Thing` feature to auto create your items, the following will need minimal editing to change the camera name from `BabyCam` to what you choose to name the camera.
```
Comment (fwolter) [inline bundles/org.openhab.binding.ipcamera/README.md:698]:
```suggestion
If you use the `Create Equipment from Thing` feature to auto create your items, the following will need minimal editing to change the camera name from `BabyCam` to what you choose to name the camera.
```

File: bundles/org.openhab.binding.ipcamera/src/main/resources/OH-INF/binding/binding.xml [hunk 2]
```diff
@@ -4,6 +4,6 @@
 	xsi:schemaLocation="https://openhab.org/schemas/binding/v1.0.0 https://openhab.org/schemas/binding-1.0.0.xsd">
 
 	<name>IpCamera Binding</name>
-	<description>This binding helps you to use IP Cameras in Openhab 2.</description>
+	<description>This binding is useful if you wish to use IP cameras in openHAB.</description>
```
Comment (fwolter) [inline bundles/org.openhab.binding.ipcamera/src/main/resources/OH-INF/binding/binding.xml:7]: "openHAB" don't need to be part of the description. Here is another suggestion:
```suggestion
	<description>This binding interfaces IP cameras of various vendors via open protocols.</description>
```

**Review by fwolter — CHANGES_REQUESTED** [review]
Please note that the title of this PR will appear in the changelog. You could express it a bit more user related. You could also include the word "refactor", to show that you made changes that don't affect the user.

If I see correctly, the only change, the user will see are the `advanced`s and some texts.

File: bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/StreamServerHandler.java [hunk 3]
```diff
@@ -177,19 +177,13 @@ public void channelRead(@Nullable ChannelHandlerContext ctx, @Nullable Object ms
...
                     incomingJpeg = new byte[recievedBytes + content.content().capacity()];
-
-                    for (; index < temp.length; index++) {
-                        incomingJpeg[index] = temp[index];
-                    }
-                }
-                for (int i = 0; i < content.content().capacity(); i++) {
-                    incomingJpeg[index++] = content.content().getByte(i);
+                    System.arraycopy(temp, 0, incomingJpeg, 0, temp.length);
+                    System.arraycopy(content.content(), 0, incomingJpeg, recievedBytes, content.content().capacity());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/StreamServerHandler.java:186]: `content.content()` isn't an array type, so I don't think `System.arraycopy` would work as written.
```suggestion
                    System.arraycopy(content.content().array(), 0, incomingJpeg, recievedBytes, content.content().capacity());
```

**Review by cpmeister — CHANGES_REQUESTED** [review]

**Review by cpmeister — APPROVED** [review]

**Review by fwolter — APPROVED** [review]
LGTM. There are three `AvoidScheduleAtFixedRateCheck` warnings left. Can you fix these (in another PR)?

**Pull request merged**